omisego-sdk/src/main/java/co/omisego/omisego/model/transaction/request/TransactionRequest.kt

```diff
@@ -0,0 +1,33 @@
+package co.omisego.omisego.model.transaction.request
+
+/*
+ * OmiseGO
+ *
+ * Created by Phuchit Sirimongkolsathien on 27/3/2018 AD.
+ * Copyright © 2017-2018 OmiseGO. All rights reserved.
+ */
+
+import android.graphics.Bitmap
+import co.omisego.omisego.model.MintedToken
+import co.omisego.omisego.qrcode.QRGenerator
+
+/**
+ * Represents a transaction request
+ * 
+ */
+data class TransactionRequest(
+    val id: String,
+    val type: TransactionRequestType,
+    val mintedToken: MintedToken,
+    val amount: Double?,
+    val address: String?
+) {
+
+    /**
+     * Generates an QR bitmap containing the encoded transaction request id
+     *
+     * @param size the desired image size
+     * @return An QR image if the transaction request was successfully encoded
+     */
+    fun generateQRCode(size: Int = QRGenerator.DEFAULT_SIZE): Bitmap = QRGenerator().generate(id, size)
```

Comment:
I think it might be a bad idea to put this here, basically injecting logic in a model class 🤔 
What do you think about having 

```
val transactionRequest = ...
QRGenerator().generate(transactionRequest)
```

and maybe if you want something like `val qrPayload: String get() = id` in this class ?

You can also add an extension in the companion of QRGenerator like

```kotlin
companion object {
    fun TransactionRequest.generateQRCode(size: Int = QRGenerator.DEFAULT_SIZE): Bitmap = QRGenerator().generate(id, size)
}
```

Usage would be the same, but the logic would be contained in the `QRGenerator` class instead of being leaked in the model class which shouldn't know anything about this.
Also will be very useful if later you decide to replace the implementation with another lib etc 😄

Reply:
It is just a convenient method like extension function to the `TransactionRequest`. I think it's the same reason why we add `showToast` method to the `Context` class 😄, isn't it?

Reply:
Oh, I got it. I just missed your point 🙏🏻that makes sense now.  Will update it 👍

Reply:
But if we put it in the `Companion object` then it will add more complex to use our extension?

instead of

```kotlin
transactionRequest.generateQRCode()
```

to something like

```kotlin
QRGenerator.run {
    transactionRequest.generateQRCode()
}
```

So this method may be harder to be discovered? because it doesn't appear in the `autocompletion` anymore when typing `transactionRequest.` 😓

Reply:
I don't think you need to run it in a `QRGenerator` context if it's in the companion no?
Worst case you put it in the same file, just not in the companion, and auto-completion should work there 🤔

Reply:
I've tried, it needs to be run in the `QRGenerator` context.

Reply:
Daymn, I learned something new 😄